### PR TITLE
continual building timetable #51

### PR DESCRIPTION
**Pull request by davidm1766**
- timetable for subject is now building continously (adding new discovered block)
-fixed bug in converter during converting

### CODE REVIEW

File: Backend/CourseService.cs [hunk 1]
```diff
@@ -33,21 +33,35 @@ public CourseService(IMongoDatabase database)
 
         public async Task<Course> FindByNameAsync(string name)
         {
-            return await _courseCollection.Find(x => x.CourseName.Equals(name.ToUpper())).FirstOrDefaultAsync();
+            return await _courseCollection.Find(x => x.CourseName.Equals(name)).FirstOrDefaultAsync();
         }
 
-        public async Task<Guid> GetOrAddNotExistsCourseId(string courseName, ICourseService courseServ, ISchoolScheduleProxy proxy)
+        public async Task<Guid> GetOrAddNotExistsCourseId(string courseName, ICourseService courseServ, Block bl)
```
Comment (marektavac) [inline Backend/CourseService.cs]: What is it b1?

File: Backend/CourseService.cs [hunk 2]
```diff
@@ -33,21 +33,35 @@ public CourseService(IMongoDatabase database)
...
+                    //is is neccessary to add it into timetable
+                    course.Timetable.Blocks.Add(bl);
+                    await courseServ.Update(course);
+                }
+            }
             return course.Id;
 
         }
+
+        public async Task Update(Course course)
```
Comment (marektavac) [inline Backend/CourseService.cs]: UpdateAsync

File: Entities/Timetable.cs [hunk 3]
```diff
@@ -12,5 +12,17 @@ public Timetable()
...
         }
+
+        public bool ContainsBlock(Block bl)
+        {
+            foreach (var blck in Blocks)
+            {
+                if(blck.IsSameAs(bl))
+                {
+                    return true;
+                }
```
Comment (marektavac) [inline Entities/Timetable.cs]: return blck,IsSameAs(b1);

File: Backend/CourseService.cs (hunk 1 above)
Comment (davidm1766) [reply Backend/CourseService.cs]: I rename it

File: Backend/CourseService.cs (hunk 2 above)
Comment (davidm1766) [reply Backend/CourseService.cs]: OK

File: Entities/Timetable.cs (hunk 3 above)
Comment (davidm1766) [reply Entities/Timetable.cs]: I cannot do that because I if I do that I will return first false

**Comment by marektavac** [comment]
@davidm1766 , You can delete branch

**Pull request merged**